File: src/main/java/com/morak/performancetracker/aop/persistence/StatementPointcut.java
package com.morak.performancetracker.aop.persistence;

import java.lang.reflect.Method;
import org.springframework.aop.support.StaticMethodMatcherPointcut;

public class StatementPointcut extends StaticMethodMatcherPointcut {

    private static final String EXECUTION_PREFIX = "execute";

    @Override
    public boolean matches(Method method, Class<?> targetClass) {
        return method.getName().contains(EXECUTION_PREFIX);
    }
}

File: src/main/kotlin/com/morak/performancetracker/aop/persistence/StatementPointcut.kt
package com.morak.performancetracker.aop.persistence

import org.springframework.aop.support.StaticMethodMatcherPointcut
import java.lang.reflect.Method

class StatementPointcut : StaticMethodMatcherPointcut() {
    override fun matches(method: Method, targetClass: Class<*>?): Boolean {
        return method.name.contains(EXECUTION_PREFIX)
    }

    companion object {
        private const val EXECUTION_PREFIX = "execute"
    }
}
